### PR TITLE
621 parsing issue objects cleanup

### LINKED ISSUE #621: Cleanup parsing issue objects
1) Txm event id in database in table parsing_issue should be required
2) ParsingIssue is used for two purposes: a temporary object  created when an issue in parsing occurs from which real parsing issue is created higher in the code, when all data is available. This is not good, split it in two objects, one with all fields required, other with just the fields that are required now, that will be used as the temporary object.
3) ParsingIssuePublicDTO is not correct, confirmed_at and confirmed by does not make sense there as it is (check it) used likely only when data are being sent to the app from external sources and in such case the issues are brand new and cannot have any confirmation. Moreover txm_event_name and medical_id should be required

### PR DESCRIPTION
**Pull request by krllstdn**
Closes #621

### CODE REVIEW

File: txmatching/utils/hla_system/hla_transformations/get_mfi_from_multiple_hla_codes.py [hunk 1]
```diff
@@ -113,4 +111,4 @@ def get_mfi_from_multiple_hla_codes(mfis: List[int],
             )
         )
 
-    return parsing_issues, relevant_mean
+    return parsing_issues_temp, relevant_mean
```
Comment (kristinagalik) [inline txmatching/utils/hla_system/hla_transformations/get_mfi_from_multiple_hla_codes.py:114]: missing newline at the end of the file

File: txmatching/patients/hla_functions.py [hunk 2]
```diff
@@ -171,4 +172,4 @@ def number_of_antigens_is_insufficient_in_high_res(recipient_antibodies: List[HL
 
     if len(recipient_antibodies) < SUFFICIENT_NUMBER_OF_ANTIGENS_IN_HIGH_RES:
         return True
-    return False
+    return False
```
Comment (kristinagalik) [inline txmatching/patients/hla_functions.py:175]: missing newline

File: txmatching/database/sql_alchemy_schema.py [hunk 3]
```diff
@@ -264,4 +270,4 @@ class UploadedFileModel(db.Model):
         server_default=func.now(),
         onupdate=func.now()
     )
-    deleted_at = Column(DATETIME(timezone=True), nullable=True)
+    deleted_at = Column(DATETIME(timezone=True), nullable=True)
```
Comment (kristinagalik) [inline txmatching/database/sql_alchemy_schema.py:273]: missing newline(pls check all the modified files)

File: txmatching/data_transfer_objects/hla/parsing_issue_swagger.py [hunk 4]
```diff
@@ -1,39 +1,26 @@
+
```
Comment (kristinagalik) [inline txmatching/data_transfer_objects/hla/parsing_issue_swagger.py:1]: this empty line shouldnt be here

**Review by kristinagalik — CHANGES_REQUESTED** [review]
maybe you copied contents of some files but most of them are missing newlines at the end

other thing is that you are at this point unable to build frontend - i know its frontend but it shouldnt be too complicated, so if you want you can try to fix it by yourself :) if you struggle, just message me or i can do it

File: txmatching/web/frontend/src/app/generated/model/patientUploadSuccessResponseGenerated.ts [hunk 5]
```diff
@@ -21,6 +21,7 @@ export interface PatientUploadSuccessResponseGenerated {
      * Errors and warnings that occurred in HLA codes parsing
      */
     parsing_issues: Array<ParsingIssuePublicGenerated>;
+    // parsing_issues: Array<ParsingIssuePublicGenerated>;
```
Comment (kristinagalik) [inline txmatching/web/frontend/src/app/generated/model/patientUploadSuccessResponseGenerated.ts:24]: this looks hard coded... maybe regenerate swagger files?

**Review by kristinagalik — CHANGES_REQUESTED** [review]
just a small comment, plus there are some formatting imperfections, but for frontend, ask @NotLubo for reviews rather than me

File: txmatching/web/frontend/src/app/services/patient/patient.service.ts [hunk 6]
```diff
@@ -24,9 +24,11 @@ import { fromDonorEditableToUpdateGenerated } from '@app/parsers/to-generated/do
 import { fromRecipientEditableToUpdateGenerated } from '@app/parsers/to-generated/recipient.parsers';
 import { fromPatientsEditableToInGenerated } from '@app/parsers/to-generated/patientPair.parsers';
 import { ParsingIssue } from '@app/model/ParsingIssue';
-import { parseParsingIssue } from '@app/parsers/parsingIssue.parsers';
-import { parseParsingIssueConfirmation } from '@app/parsers/parsingIssueConfirmation.parsers';
+import { parseParsingIssue} from '@app/parsers/parsingIssue.parsers';
+import {parseParsingIssuePublic} from '@app/parsers/parsingIssuePublic.parsers';
+// import { parseParsingIssue } from '@app/parsers/parsingIssueConfirmation.parsers';
```
Comment (NotLubo) [inline txmatching/web/frontend/src/app/services/patient/patient.service.ts:29]: remove comment

File: txmatching/web/frontend/src/app/services/patient/patient.service.ts [hunk 7]
```diff
@@ -24,9 +24,11 @@ import { fromDonorEditableToUpdateGenerated } from '@app/parsers/to-generated/do
 import { fromRecipientEditableToUpdateGenerated } from '@app/parsers/to-generated/recipient.parsers';
 import { fromPatientsEditableToInGenerated } from '@app/parsers/to-generated/patientPair.parsers';
 import { ParsingIssue } from '@app/model/ParsingIssue';
-import { parseParsingIssue } from '@app/parsers/parsingIssue.parsers';
-import { parseParsingIssueConfirmation } from '@app/parsers/parsingIssueConfirmation.parsers';
+import { parseParsingIssue} from '@app/parsers/parsingIssue.parsers';
+import {parseParsingIssuePublic} from '@app/parsers/parsingIssuePublic.parsers';
```
Comment (NotLubo) [inline txmatching/web/frontend/src/app/services/patient/patient.service.ts:28]: lines 28 and 27 formatting

**Review by NotLubo — CHANGES_REQUESTED** [review]
Minor formatting issues and unremoved leftover code. Good job :)

File: txmatching/database/db_migrations/0033.make-txm-event-id-required-in-parsing-issue.sql [hunk 8]
```diff
@@ -0,0 +1,8 @@
+--
+-- file: txmatching/database/db_migrations/0033.make-txm-event-id-required-in-parsing-issue.sql
+-- depends: 0032.add-confirmation-columns-to-parsing-issue-model
+--
+
+ALTER TABLE parsing_issue ALTER COLUMN txm_event_id SET NOT NULL;
+ALTER TABLE parsing_issue ALTER COLUMN hla_code_or_group SET NOT NULL;
+ALTER TABLE parsing_issue ADD CHECK ((donor_id IS NOT NULL) OR (recipient_id IS NOT NULL));
```
Comment (kristinagalik) [inline txmatching/database/db_migrations/0033.make-txm-event-id-required-in-parsing-issue.sql:8]: newline missing

File: txmatching/database/sql_alchemy_schema.py [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import dataclasses
 from typing import List
 
-from sqlalchemy.schema import Column, ForeignKey, UniqueConstraint
+from sqlalchemy.schema import Column, ForeignKey, UniqueConstraint, CheckConstraint
```
Comment (kristinagalik) [inline txmatching/database/sql_alchemy_schema.py:4]: alphabetical order

**Review by kristinagalik — CHANGES_REQUESTED** [review]
just small be changes, otherwise backend ok :)

**Review by NotLubo — APPROVED** [review]

**Review by kubantjan — APPROVED** [review]

**Pull request merged**